runtime/floats.c

```diff
@@ -18,7 +18,9 @@
 /* The interface of this file is in "caml/mlvalues.h" and "caml/alloc.h" */
 
 /* Needed for uselocale */
+#if !defined(__SUNPRO_C)
 #define _XOPEN_SOURCE 700
+#endif
```

Comment:
It is strange that this (and the other disabling of XOPEN_SOURCE like that) would be necessary. I looked around the web and it looks rather similar to [this issue](https://mail-archives.apache.org/mod_mbox/stdcxx-dev/201209.mbox/%3C505D43C8.3080205@hates.ms%3E). I think it would be nice to have a comment in the source as to why __SUNPRO_C is handled specially here.

Reply:
Unfortunately this is a problem of old standard library of `Sun PRO C`. I do not know a better way to fix it. We can add a comment about it.

Reply:
Thanks **Xavier Leroy** I found out that it is not necessary and can be substituted with `-D_XPG6` compiler option. Will remove.